app/screens/apps_form/apps_form_component.tsx

```diff
@@ -125,27 +126,15 @@ function AppsFormComponent({
     const theme = useTheme();
     const style = getStyleFromTheme(theme);
 
-    useEffect(() => {
-        const unsubscribe = Navigation.events().registerComponentListener({
-            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
-                switch (buttonId) {
-                    case CLOSE_BUTTON_ID:
-                        close();
-                        break;
-                    case SUBMIT_BUTTON_ID: {
-                        if (!submitting) {
-                            handleSubmit();
-                        }
-                        break;
-                    }
-                }
-            },
-        }, componentId);
-        return () => {
-            unsubscribe.remove();
-        };
+    const onHandleSubmit = useCallback(() => {
+        if (!submitting) {
+            handleSubmit();
+        }
     }, [serverUrl, componentId, submitting]);
 
+    useNavButtonPressed(CLOSE_BUTTON_ID, componentId, close, [close]);
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleSubmit, [onHandleSubmit]);
```

Comment:
not a fan, this subscribes multiple listeners when one is more than enough but this makes the code cleaner for sure and I really doubt one more listener makes the difference to keep this out

Reply:
Yes.